typescript/looker/rtl/extensionTransport.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2019 Looker Data Sciences, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+import {
+    SDKResponse,
+    ITransport,
+    ITransportSettings,
+    HttpMethod
+} from './transport'
+import {IHostConnection} from './extensionSdk'
+
+export class ExtensionTransport implements ITransport {
+  constructor(
+    private options: ITransportSettings,
+    private hostConnection: IHostConnection
+  ) {
+    this.options = options
+    this.hostConnection = hostConnection
+  }
+
+  async request<TSuccess, TError>(
+    method: HttpMethod,
+    path: string,
+    queryParams?: any,
+    body?: any,
+    authenticator?: any,
+    options?: Partial<ITransportSettings>,
+  ): Promise<SDKResponse<TSuccess, TError>> {
+    let params: any
+    if (queryParams) {
+        params = {}
+        Object.keys(queryParams).forEach(key => {
+            if (queryParams[key]) {
+                params[key] = queryParams[key]
+            }
```

Comment:
are your parameter passing constraints intentionally different than `encodeParams` in `transport.ts`? That routine is a bit more hardened, with encoding of query parameters, etc, and explicit checks only against `undefined` because this loop will omit parameter values you may not actually want to be omitted if they evaluate to "falsey."

Reply:
I've actually been thinking about this code and have come to the conclusion that it should probably be in helltool. This came about in the rush to get things ready for JOIN. Basically the host was failing if it received queryParams with undefined values. This was a quick and dirty to get Wil unblocked.

Reply:
Moved to helltool and removed from here